contracts/FlashBuy.sol

```diff
@@ -201,31 +81,39 @@ contract FlashBuy is IERC3156FlashBorrower, OwnableUpgradeable {
             uint256 auctionId,
             address collateral,
             uint256 collateralAm,
-            uint256 maxPrice
-        ) = abi.decode(data, (Action, uint256, address, uint256, uint256));
+            uint256 maxPrice,
+            uint256 slip,
+            address collateralReal,
+            bytes memory path
+        ) = abi.decode(data, (Action, uint256, address, uint256, uint256, uint256, address, bytes));
         require(action == Action.NORMAL, "such action is not implemented");
         uint256 tokenBalance = IERC20(token).balanceOf(address(this));
         require(tokenBalance >= amount, "borrow amount not received");
 
-        auction.buyFromAuction(
+        uint256 before = IERC20(collateralReal).balanceOf(address(this));
+
+        interaction.buyFromAuction(
             collateral,
             auctionId,
             collateralAm,
             maxPrice,
             address(this)
         );
-        uint256 minOut = amount + fee;
 
-        address[] memory path = new address[](2);
-        path[0] = collateral;
-        path[1] = token;
-        dex.swapExactTokensForTokens(
-            collateralAm,
-            minOut,
-            path,
-            address(this),
-            block.timestamp + 300
-        );
+        uint256 amountIn = IERC20(collateralReal).balanceOf(address(this)) - before;
+        IERC20(collateralReal).approve(address(dex), amountIn);
+
+        uint256 currentPrice = interaction.collateralPrice(collateral);
+        uint256 amountOut = (amountIn * currentPrice) / 1e18;
+        uint256 amountOutMin = (amountOut * (MAX_SLIPE - slip)) / MAX_SLIPE;
```

Comment:
`amountOutMint` should bigger than minOut, otherwise throw an error

Reply:
<img width="768" alt="image" src="https://github.com/user-attachments/assets/da1089c0-c476-45fe-bce7-4d8f46397b3d">

dex will ensure amountOut >= amountOutMin